test/integration/scheduler/podsready/scheduler_test.go

```diff
@@ -249,47 +249,60 @@ var _ = ginkgo.Describe("SchedulerWithWaitForPodsReady", func() {
 			ginkgo.By("create the 'prod' workload")
 			prodWl := testing.MakeWorkload("prod", ns.Name).Queue(prodQueue.Name).Request(corev1.ResourceCPU, "2").Obj()
 			gomega.Expect(k8sClient.Create(ctx, prodWl)).Should(gomega.Succeed())
+
 			ginkgo.By("checking the 'prod' workload is admitted")
 			util.ExpectWorkloadsToHaveQuotaReservation(ctx, k8sClient, prodClusterQ.Name, prodWl)
 			util.ExpectQuotaReservedWorkloadsTotalMetric(prodClusterQ, 1)
 			util.ExpectAdmittedWorkloadsTotalMetric(prodClusterQ, 1)
-			ginkgo.By("exceed the timeout for the 'prod' workload")
-			time.Sleep(podsReadyTimeout)
+			util.AwaitWorkloadEvictionByPodsReadyTimeoutAndSetRequeuedCondition(ctx, k8sClient, client.ObjectKeyFromObject(prodWl), podsReadyTimeout)
```

Comment:
No strong view here, but I prefer not to mix too many responsibilities into a single function, the name looks long. Maybe we can have two functions: `AwaitWorkloadEvictionByPodsReadyTimeout` and `SetRequeuedCondition`?

Reply:
That sounds reasonable. Let me split this helper into two ones.
Thanks!

Reply:
Done.